generator/testdata/google/example/library/v1/library_service_stub.gapic.h.baseline

```diff
@@ -38,4 +38,13 @@ class LibraryServiceStub {
 
 }; // LibraryServiceStub
 
+std::unique_ptr<LibraryServiceStub> CreateLibraryServiceStub(
```

Comment:
Can we:
 - add a factory method that takes no arguments, and uses GoogleDefaultCredentials()
 - Remove some of these. I don't think we need them all, they will just make users confused more than anything. My inclination is to keep the one that accepts a channel, since that is a sensible thing to share around. But I think it might also be reasonable to not provide any methods except the one with 0 args, since we don't really know which additional factory method will be most useful. WDYT?

Reply:
+1 for a no-arg factory.

The raw stub factory seems to be of limited use: our generated Stub class is (currently by coincidence and in future by design) thread safe, and the intention is that multiple clients interact with the generated Stub instead of the raw grpc stub.

I don't think we really want a Channel based factory. That forces the user to pass in the correct URI, which extreme customization notwithstanding defeats the purpose of the client library.
The two things users might reasonably want to customize are ChannelArguments and Credentials, both of which have reasonable default values that should cover the majority of use cases. If we allow all combinations (ChannelArguments, Credentials) x (default, user provided) that gives us four factories.

Reply:
I think that's a good point about the channel. I'm then inclined to just stick with the no-arg constructor, or perhaps add one more for creds. I wouldn't add ChannelArguments at this stage - we can add it later if needed, but it is pretty obscure. Let's prefer less for now, and then add more when we need it.

Reply:
Sounds good. Removing the grpc_stub and ChannelInterface factories and only providing a default factory and one that takes Credentials.